### PR TITLE
Adding signupTeams support

### PR DESCRIPTION
**Pull request by pnoll1**
Currently includes 1st half of support up to rendering signupTeams on registration page

### CODE REVIEW

File: routes/actions.js [hunk 1]
```diff
@@ -343,25 +344,52 @@ router.post('/register/:code', function(req, res, next) {
...
     titleKey: 'register',
     pageErrors,
     formValues: formInfo ? formInfo.formValues : undefined,
     questionCaptcha: forms.getQuestionCaptcha('register'),
     illegalUsernameCharactersReadable: User.options.illegalCharsReadable,
     scripts: ['register.js'],
     inviteCode: code,
-    signupLanguage: req.query.signupLanguage || req.body.signupLanguage
-  }, {
+    signupLanguage: req.query.signupLanguage || req.body.signupLanguage,
```
Comment (pnoll1) [inline routes/actions.js]: Possible unintended change need to investigate

File: routes/actions.js [hunk 2]
```diff
@@ -343,25 +344,52 @@ router.post('/register/:code', function(req, res, next) {
...
-
-  render.template(req, res, 'register', {
+  const signupTeams = req.query.signupTeams;
+  // checks if signupTeams exist, valid teams are passed to render
+  if (signupTeams) {
+    let signupTeamsArray = signupTeams.split(' ');
+    let validSignupTeams = [];
+    let promise_array = [];
+    for (let i = 0; i < signupTeamsArray.length; i++) {
+      let team = slugs.resolveAndLoadTeam(req, res, signupTeamsArray[i])
```
Comment (pnoll1) [inline routes/actions.js]: It may be more appropriate to use a method from team-provider.js as you originally suggested.  I used slugs instead because team-provider methods were returning errors about missing arguments when I tried to pass the slug alone.

File: views/register.hbs [hunk 3]
```diff
@@ -29,6 +29,18 @@
             <input id="email" name="email" type="text" data-auto-trim placeholder="{{{__ "enter email"}}}" value="{{formValues.email}}" size="30" class="pure-input-1 login-form-input">
         </div>
 
+        {{#if signupTeamsTransfer}}
+        <div class="pure-control-group">
+          <h4>Teams suggested by invite link, check box to join</h4>
```
Comment (pnoll1) [inline views/register.hbs]: May want to add explanation of what teams are or link to explanation of teams

File: routes/actions.js [hunk 4]
```diff
@@ -267,6 +268,7 @@ if (!config.requireInviteLinks) {
           if (error) {
             debug.error({ req, error });
           }
+          setSignupLanguage(req, res);
```
Comment (eloquence) [inline routes/actions.js]: This seems unrelated to the team feature?

File: routes/actions.js [hunk 5]
```diff
@@ -341,25 +344,52 @@ router.post('/register/:code', function(req, res, next) {
 
 });
 
-function sendRegistrationForm(req, res, formInfo) {
+async function sendRegistrationForm(req, res, formInfo) {
```
Comment (eloquence) [inline routes/actions.js]: Marking a function `async` means that it will return a promise -- and all invocations of the function have to be changed to account for that. This is probably not what you want to do here.

File: views/register.hbs (hunk 3 above)
Comment (eloquence) [reply views/register.hbs]: We'll also need to internationalize the text as we get closer to merge

File: routes/actions.js (hunk 5 above)
Comment (pnoll1) [reply routes/actions.js]: My understanding is that I need the function to be async to allow usage of await within it.  It's in one the promise tutorials you sent: [ponyfoo async await](https://ponyfoo.com/articles/understanding-javascript-async-await#using-async-await).

File: routes/actions.js (hunk 4 above)
Comment (pnoll1) [reply routes/actions.js]: How do you want to handle this?  I could delete this PR, rollback changes, rebase and put changes in a feature branch.

File: routes/actions.js (hunk 5 above)
Comment (eloquence) [reply routes/actions.js]: That's true, but it changes the return value to a promise, so it has consequences for all callers of the function. You can update all callers to account for this fact, refactor the function into smaller ones, or avoid the use of `await`.

Comment (eloquence) [reply routes/actions.js]: Poking a bit more at this particular case, it looks like the `async` change is probably safe here, since the callers don't depend on the return value of the function, which is always `undefined`. If it passes tests and the PR is complete on your end, I'll poke a bit more to see if there are any side effects of this change.

File: routes/actions.js (hunk 4 above)
Comment (eloquence) [reply routes/actions.js]: Yes, that makes sense to me.

**Comment by pnoll1** [comment]
closing PR, will open seperate PR once changes have been split into feature branch
